### DISCUSSION TITLE
Why not FluentIcons use a classic CSS reference or Img tag?

### DISCUSSION THREAD

**Discussion by damianog** [microsoft/fluentui-blazor#455] [Q&A] [answered]
Due to my issue #446 I digged into FluentIcon code and I noticed a complex way to retrieve svg icons by httpclient and cache.

But at the end once published icons can be retrieved by a simple url path, so why not reference them with a simple css class or render them with an Img tag, and let the browser do the caching job?

This is my point of view, I actually use only Blazor Server side.

Thanks.

**Comment by dvoituron** [comment, marked as answer]
We're working on this at the moment, and we're thinking of including all icons in a lib. This will allow them to be added directly to img or in HTML code, with the main advantage of being able to use them both in web assemblies or server, while using the browser's cache.

By having the code in a .NET library, developers will be able to manipulate the SVG icons before returning them to the Web page.

The .NET Trim feature provides a lightweight library for production use.
